framework/_partials/_preview-navigation-inpages.twig

```diff
@@ -0,0 +1,71 @@
+          <p class="ecl-paragraph">Donec egestas fermentum nisl vel aliquet. Pellentesque dui massa, elementum sed laoreet at, blandit sit amet magna. Praesent ac lobortis turpis. Maecenas sem est, consequat in erat sit amet, malesuada cursus arcu. Vestibulum molestie massa nec elit luctus euismod. Nullam nec diam et augue mattis mollis. Phasellus efficitur id velit eu vestibulum. Donec ullamcorper semper porttitor. Vivamus et sagittis urna. Nunc sit amet lectus nec magna cursus vehicula. In elementum pretium pellentesque. Vivamus est metus, hendrerit eget urna venenatis, pellentesque cursus lacus. Etiam tempus justo eget velit ultrices, in tempor arcu sodales.</p>
+        </section>
+
+        <section id="inline-nav-3">
+          <h2 class="ecl-heading ecl-heading--h2">Heading 3</h2>
+          <p class="ecl-paragraph">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam id dui ac lacus interdum eleifend non ut nisl. Duis arcu turpis, tristique vitae fringilla feugiat, tincidunt ac enim. Duis tincidunt purus quis metus finibus, ut ultricies nibh mattis. Ut ultricies placerat leo ac egestas. Suspendisse fringilla augue quam. Proin velit ipsum, bibendum ut commodo non, mattis at nisi. Sed massa eros, malesuada eu ipsum non, euismod finibus massa. Etiam mi ex, fringilla quis urna non, dignissim vulputate tortor. Vivamus congue sem vel ipsum posuere vulputate. Praesent ac vestibulum sem, sed tristique ligula. Praesent non eros suscipit, sagittis quam in, commodo felis.</p>
+          <p class="ecl-paragraph">Donec egestas fermentum nisl vel aliquet. Pellentesque dui massa, elementum sed laoreet at, blandit sit amet magna. Praesent ac lobortis turpis. Maecenas sem est, consequat in erat sit amet, malesuada cursus arcu. Vestibulum molestie massa nec elit luctus euismod. Nullam nec diam et augue mattis mollis. Phasellus efficitur id velit eu vestibulum. Donec ullamcorper semper porttitor. Vivamus et sagittis urna. Nunc sit amet lectus nec magna cursus vehicula. In elementum pretium pellentesque. Vivamus est metus, hendrerit eget urna venenatis, pellentesque cursus lacus. Etiam tempus justo eget velit ultrices, in tempor arcu sodales.</p>
+        </section>
+      </div>
+    </div>
+
+    <script src="{{ '/framework/scripts/europa.js' | path }}"></script>
+    {% if _target.context._demo and _target.context._demo %}
+    <script type="text/javascript">{{ _target.context._demo.scripts | raw }}</script>
+    {% endif %}
+
+    <!-- https://github.com/dollarshaveclub/stickybits -->
+    <script src="/framework/scripts/stickybits.min.js"></script>
+    <script type="text/javascript">
```

Comment:
As discussed, I agree that we can abstract most of the logic here in a ECL script (and just have to call something like `ECL.scrollspy('.something');`) :)